contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,448 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol" as Ownable;
+
+contract SeriesIndexer is Ownable.Ownable {
+    // Counter
+    uint256 private nextSeriesID = 1;
```

Comment:
Can you use the `length` of existing series map so we don't need to maintain this counter?

Reply:
this's not possible since the mapping in solidity is a simple hash table without key management.